src/services/search-service.ts

```diff
@@ -1,18 +1,45 @@
+import { RowDataPacket } from "mysql2";
 import logger from "../utils/logger/logger";
-
-// import SQLService from "./sql-service";
+import SQLService from "./sql-service";
 
 class SearchService {
-	static async search() {
-		logger.info("Search");
+	static async search(
+		table: string,
+		key: string,
+		columns: string[]
+	): Promise<RowDataPacket[]> {
+		logger.info(
+			`SearchService invoked! Table = ${table}, Key = ${key}, Columns = ${columns}`
+		);
+
+		const whereParams = ` WHERE MATCH (${columns})`;
+		const againstParams = ` AGAINST ('${key}' IN NATURAL LANGUAGE MODE)`;
```

Comment:
Does this SQL code here create the index in real-time? Or is there additional SQL code to build the index ahead of API usage?